src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -624,6 +721,121 @@ int decryptFinal(byte[] in, int inOfs, int len,
         return len;
     }
 
+    // Note: In-place operations do not need an intermediary copy because
+    // the GHASH check was performed before the decryption.
+    int decryptFinal(ByteBuffer src, ByteBuffer dst)
+        throws IllegalBlockSizeException, AEADBadTagException,
+        ShortBufferException {
+        // Length of the input
+        ByteBuffer tag;
+        ByteBuffer ct = src.duplicate();
+        ct.mark();
+
+        ByteBuffer buffer = ((ibuffer == null || ibuffer.size() == 0) ? null :
+            ByteBuffer.wrap(ibuffer.toByteArray()));
+        int len;
+
+        if (ct.remaining() >= tagLenBytes) {
+            tag = src.duplicate();
+            tag.position(ct.limit() - tagLenBytes);
+            ct.limit(ct.limit() - tagLenBytes);
+            len = ct.remaining();
+            if (buffer != null) {
+                len += buffer.remaining();
+            }
+            ct.mark();
```

Comment:
This seems redundant given the ct.mark() call on line 732?

Reply:
Does seem redundant.. I'll remove and see how testing goes